components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/ChallengeQuestionManager.java

```diff
@@ -534,6 +540,29 @@ public ChallengeQuestion getUserChallengeQuestion(User user, String challengesUr
         return challenges;
     }
 
+    /**
+     * Get the existing answers for the challenge questions.
+     *
+     * @param user                 User
+     * @param userChallengeAnswers List of UserChallengeAnswer objects
+     * @return Existing challenge questions and answers.
+     */
+    private Map<String, String> retrieveAnsweredChallenges(User user, UserChallengeAnswer[] userChallengeAnswers)
+            throws IdentityRecoveryException {
+
+        List<String> claimsList = new ArrayList<>();
+        for (UserChallengeAnswer answer : userChallengeAnswers) {
+            claimsList.add(answer.getQuestion().getQuestionSetId().trim());
```

Comment:
Can question and questionSetID become null?

Reply:
Fixed in https://github.com/wso2-extensions/identity-governance/pull/426/commits/50aebef76b14e869930e8c7582307d9d548c109e